### PR TITLE
feat(web): add version selection option during plugin upload 

### PR DESCRIPTION
**Pull request by mulengawilfred**
Added UI functionality for users to specify whether they want to upload the 'classic' or 'visualizer' version of the plugin. Mostly UI changes based on the new design. Also upgraded yarn and ant (5.21.0), added prettier configurations and a few packages to fix typescript errors. Would need to upgrade a few more down the road. 

**Still TODO**
- Not yet able to filter plugins by version on the homepage or upload a plugin with a specified version though uploading the plugin works as normal. Would require some backend changes. 

PR ready for review.

## Release Notes

- **New Features**
  - Introduced a new `ArrowLeftOutlined` icon for enhanced UI options.
  - Added a `ButtonNavigation` component for improved plugin management navigation.
  - Implemented a new tabbed interface in `TopPageContents` for better navigation.
  - Added a new `pluginUploaded` prop in the `PluginUpload` component to manage upload status.
  - Introduced a new `ShadowCard` component for improved visual presentation.
  - Added a `VersionArea` component for selecting plugin version environments.

- **Enhancements**
  - Updated dependency versions for improved compatibility and performance.
  - Redesigned the `PackageArea`, `SettingArea`, and `TopPageContents` components for improved layout and usability.
  - Enhanced navigation experience with new tab management and state handling.
  - Improved empty state messaging in the `MyPluginsTable` component to guide users.

- **Internationalization**
  - Updated English and Japanese translation files to improve clarity and consistency in user interface text.

- **Style**
  - Added new CSS rules for Ant Design tabs to enhance visual styling and user experience.

### CODE REVIEW

File: web/src/components/molecules/TopPage/index.tsx [hunk 1]
```diff
@@ -33,10 +34,12 @@ const TopPage: React.FC<Props> = ({
   onPageChange,
   loadingPlugins,
   pageSize,
+  setCurrentVersion,
 }) => {
   return (
     <Wrapper>
-      {showBanner && <TitleBar />}
+      {/* NOTE: No banner in the new design */}
+      {/* {showBanner && <TitleBar />} */}
```
Comment (KaWaite) [inline web/src/components/molecules/TopPage/index.tsx:42]: If there is no more banner, remove this code and the TitleBar component completely.

File: web/src/components/molecules/TopPage/TopPageContents.tsx [hunk 2]
```diff
@@ -63,4 +101,11 @@ const Wrapper = styled.div`
   background: transparent;
 `;
 
+const TabsWrapper = styled.div`
+  background: #070707;
+  display: flex;
+  justify-content: center;
+  margin-bottom: 48px;
+`;
```
Comment (KaWaite) [inline web/src/components/molecules/TopPage/TopPageContents.tsx:109]: Can you double check if this component is being used as an extension in ReEarth Viz? If so, adding background color here will break the UI on the extension side (where we have theming)

File: web/src/components/molecules/TopPage/TopPageContents.tsx [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
 import { styled } from "@marketplace/theme";
+import { Version } from "@marketplace/types";
+
+type TabKeys = "0" | "1";
+
+type Tabs = {
+  key: TabKeys;
+  label: string;
+  value: Version;
+};
```
[the remark this answers is not in the document]
Comment (KaWaite) [reply web/src/components/molecules/TopPage/TopPageContents.tsx:16]: @mulengawilfred Follow this instruction.

File: web/src/components/molecules/Common/ShadowCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export type Props = {
+  children: React.ReactElement;
+};
+
+const ShadowCard: React.FC<Props> = ({ children }) => {
+  return <CustomCard>{children}</CustomCard>;
+};
+
+const CustomCard = styled(Card)`
+  background-color: #fafafa;
```
Comment (KaWaite) [inline web/src/components/molecules/Common/ShadowCard/index.tsx:13]: Same here. Need to make sure any background-color doesn't break UI in extension.

File: web/src/components/molecules/Common/PluginUpload/index.tsx [hunk 5]
```diff
@@ -33,70 +32,61 @@
...
   isLoading,
+  pluginUploaded,
   onParseFromUrl,
   onParseFromFile,
   onPluginSave,
   onRemove,
   onPublish,
   onImagesUpload,
 }) => {
+  const [currentTab, updateTab] = useState<"1" | "2" | "3">("1");
```
[the remark this answers is not in the document]
Comment (KaWaite) [reply web/src/components/molecules/Common/PluginUpload/index.tsx:43]: @mulengawilfred If possible, you should follow I think most coderabbit comments on this file. Checking if tab === "0" has literally zero intent someone reading the code can draw from it. 
Doing like coderabbit suggests, or something similar where the currentTab's value has meaning, is preferrable

File: web/src/components/molecules/Common/PluginUpload/PackageArea.tsx [hunk 6]
```diff
@@ -1,42 +1,36 @@
...
   onParsePlugin,
   onChangeGithubUrl,
 }) => {
   const t = useT();
 
   const [uploadedFile, uploadFile] = useState<UploadFile[]>();
-  const [currentRadio, changeRadio] = useState<"Upload from local" | "GitHub repository">(
-    "Upload from local",
-  );
+  const [currentRadio, changeRadio] = useState<"Upload from local" | "GitHub">("Upload from local");
```
[the remark this answers is not in the document]
Comment (KaWaite) [reply web/src/components/molecules/Common/PluginUpload/PackageArea.tsx:33]: @mulengawilfred This too is good

File: web/src/components/molecules/TopPage/index.tsx (hunk 1 above)
Comment (mulengawilfred) [reply web/src/components/molecules/TopPage/index.tsx]: Thanks Kyle. This has been removed.

File: web/src/components/molecules/TopPage/TopPageContents.tsx (hunk 2 above)
Comment (mulengawilfred) [reply web/src/components/molecules/TopPage/TopPageContents.tsx:111]: Thanks for the catch. Going to conditionally render these tabs to not render if the marketplace is used as an extension.

File: web/src/components/molecules/Common/PluginUpload/PackageArea.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/components/molecules/Common/PluginUpload/PackageArea.tsx]: Thanks. I have added these changes.

File: web/src/components/molecules/Common/PluginUpload/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/components/molecules/Common/PluginUpload/index.tsx]: done

File: web/src/components/molecules/TopPage/TopPageContents.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/components/molecules/TopPage/TopPageContents.tsx]: done

File: web/src/i18n/translations/ja.yml [hunk 7]
```diff
@@ -35,22 +35,23 @@ My Plugins: マイプラグイン
 Log Out: ログアウト
 Upload new plugin: 新しいプラグインをアップロード
 Log in: ログイン
-Plugins List: プラグイン一覧
-New Plugin: 新規プラグインの追加
+Prev: ''
+Next: ''
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/i18n/translations/ja.yml]: done

File: web/src/components/molecules/Common/PluginUpload/PackageArea.tsx [hunk 8]
```diff
@@ -72,56 +70,83 @@
...
-            />
-            <p>{t("Please set your repository as public respository.")}</p>
-          </>
-        )}
-      </Space>
-    </Wrapper>
+    <ShadowCard>
+      <ContentWrapper>
+        <Space direction="vertical" size={"large"}>
+          <ContentText>Package</ContentText>
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/components/molecules/Common/PluginUpload/PackageArea.tsx]: done

File: web/src/i18n/translations/ja.yml [hunk 9]
```diff
@@ -83,7 +85,9 @@
 Published Desc: 公開日降順
 search text: キーワードを入力
 Liked: お気に入り済
-Details: 詳細
+Classic: ''
+Visualizer: ''
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/i18n/translations/ja.yml]: done

File: web/src/i18n/translations/ja.yml [hunk 10]
```diff
@@ -35,37 +35,39 @@ My Plugins: マイプラグイン
...
 Please set your repository as public respository.: レポジトリをPublicにするのをお忘れなく。
 Image uploaded successfully.: 画像がアップロードされました。
 Image upload failed.: 画像のアップロードに失敗しました。
-Back: 戻る
 Plugin Name: プラグイン名
 Version: バージョン
+Description: 詳細
 Images: 画像
 Upload Image: 画像の追加
+Which environment is your plugin developed for ?: ''
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/i18n/translations/ja.yml]: done

File: web/src/components/molecules/Common/ShadowCard/index.tsx (hunk 4 above)
Comment (mulengawilfred) [reply web/src/components/molecules/Common/ShadowCard/index.tsx]: Thanks for spotting this. Changed it to `transparent`

File: web/src/components/molecules/Common/ShadowCard/index.tsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+const ShadowCard: React.FC<Props> = ({ children }) => {
+  return <CustomCard>{children}</CustomCard>;
+};
+
+const CustomCard = styled(Card)`
+  background-color: transparent;
+  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
+`;
```
Comment (airslice) [inline web/src/components/molecules/Common/ShadowCard/index.tsx:15]:
```suggestion
const ShadowCard = styled(Card)`
  background-color: transparent;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
`;
```
Seems like we can use styled component directly here?

File: web/src/components/molecules/MyPluginsPage/MyPluginsTable.tsx [hunk 12]
```diff
@@ -124,21 +125,33 @@ const MyPluginsTable: React.FC<Props> = ({
     [plugins],
   );
 
+  const renderEmpty = () => {
+    return (
+      <RenderEmptyWrapper>
+        <RenderEmptyText>
+          {t("Click the  “ New Plugin ” button to upload your first Plugin")}
```
Comment (airslice) [inline web/src/components/molecules/MyPluginsPage/MyPluginsTable.tsx:132]:
```suggestion
          {t('Click the "New Plugin" button to upload your first Plugin')}
```
“” seems to be some special characters.

File: web/src/components/organisms/Top/index.tsx [hunk 13]
```diff
@@ -1,19 +1,22 @@
...
   accessToken?: string;
+  version?: Version | undefined;
   onPluginSelect?: (pluginId: string) => void;
 };
 
-const Top: React.FC<Props> = ({ showBanner, accessToken, onPluginSelect }) => {
+const Top: React.FC<Props> = ({ accessToken, onPluginSelect, version }) => {
   const [searchText, updateSearchText] = useState<string>("");
   const [isFavSelected, toggleLiked] = useState<boolean>(false);
+  const [_currentVersion, setCurrentVersion] = useState<Version>("classic");
```
Comment (airslice) [inline web/src/components/organisms/Top/index.tsx:17]:
```suggestion
  const [_currentVersion, setCurrentVersion] = useState<Version>(version ?? "classic");
```

File: web/src/types/index.ts [hunk 14]
```diff
@@ -0,0 +1,7 @@
+export enum Tabs {
```
Comment (airslice) [inline web/src/types/index.ts:1]: Naming `Tabs` seems a little bit too generic, what do you think?

Comment (mulengawilfred) [reply web/src/types/index.ts]: Thanks. I have renamed it to `TabsType` as we have a component called `Tabs`

File: web/src/components/organisms/Top/index.tsx (hunk 13 above)
Comment (mulengawilfred) [reply web/src/components/organisms/Top/index.tsx]: Nice. Added this

File: web/src/components/molecules/MyPluginsPage/MyPluginsTable.tsx (hunk 12 above)
Comment (mulengawilfred) [reply web/src/components/molecules/MyPluginsPage/MyPluginsTable.tsx]: Yes, I copied them from the design but replaced them as suggested.

File: web/src/components/molecules/Common/ShadowCard/index.tsx (hunk 11 above)
Comment (mulengawilfred) [reply web/src/components/molecules/Common/ShadowCard/index.tsx]: Sweet. I didn't know that. Added

File: web/src/components/molecules/Common/PluginUpload/SettingArea.tsx [hunk 15]
```diff
@@ -43,47 +42,50 @@ const SettingArea: React.FC<Props> = ({ pluginName, version, onBack, handleUploa
...
 `;
 
-// const StyledTextArea = styled(TextArea)`
-//   width: 100%;
-// `;
+const StyledDiv = styled.div`
+  min-height: 100px;
+  border-width: 1px;
+  border-style: solid;
+  padding: 6px 8px; 8px; 8px;
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/components/molecules/Common/PluginUpload/SettingArea.tsx]: done

**Review by KaWaite — APPROVED** [review]

**Review by airslice — APPROVED** [review]

**Pull request merged**